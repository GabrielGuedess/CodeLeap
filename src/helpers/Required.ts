export type Required<T> = {
  [P in keyof T]-?: T[P];
};
