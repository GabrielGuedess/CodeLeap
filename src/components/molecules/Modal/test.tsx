import { renderWithTheme } from 'utils/tests/helpers';

import { Modal } from '.';

describe('<Modal />', () => {
  it('should render the heading', () => {
    // Arrange
    const { container } = renderWithTheme(
      <Modal trigger={<h1>click</h1>} open>
        <h2>Modal</h2>
      </Modal>,
    );

    // Assert
    expect(container.firstChild).toMatchSnapshot();
  });
});
