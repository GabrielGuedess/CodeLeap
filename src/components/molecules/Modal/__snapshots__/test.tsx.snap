// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render the heading 1`] = `
.c0 {
  background: transparent;
  border: none;
}

.c0:focus {
  box-shadow: #B8BEF3 0px 0px 0px 1.5px;
}

<button
  aria-controls="radix-:r0:"
  aria-expanded="true"
  aria-haspopup="dialog"
  class="c0"
  data-state="open"
  type="button"
>
  <h1>
    click
  </h1>
</button>
`;
